models/issue.go

```diff
@@ -1498,3 +1507,46 @@ func updateIssue(e Engine, issue *Issue) error {
 func UpdateIssue(issue *Issue) error {
 	return updateIssue(x, issue)
 }
+
+// UpdateIssueDeadline does what it says
```

Comment:
Please correct comment...

Reply:
How? Do you want me to write something like `UpdateIssueDeadline updates an issue deadline`?

Reply:
That's better. And does it really update only deadline? I think it updates all given fields right now.

Reply:
It also creates comments. I am not sure if it is good. Maybe better to add comments directly in udate/remove deadline function?

Reply:
> That's better. And does it really update only deadline? I think it updates all given fields right now.

Well, it updates all fields to its old value + issueDeadline to the new value. Although I think xorm does some magic here and only updates the fields which did not change.

> It also creates comments. I am not sure if it is good. Maybe better to add comments directly in udate/remove deadline function?

This is the function which updates/removes the deadline...

Reply:
So the only diff between this function and `updateIssue` is that this func creates comments. Function doesn't know what are the fields of `issue`, who and why set it to given values, only caller knows it (function doesn't know context about when it is called, I can call it from anywhere). If I look at implementation, more meaningful explanation is "UpdateIssueDeadline creates deadline comment and updates issue".

I don't think Xorm is so magical. As I know, it can ignore default values, but doesn't know what is updated. But you can check it by turning on sql logging so we can be sure about that.

Sry, I should be more concrete about "update/remove" functions. I meant `UpdateDeadline` and `RemoveDeadline`. But as there is not `xorm.Session` it can not be called from there directly.